File: app/src/main/java/com/an/trailers/data/local/converter/TvListTypeConverter.java
package com.an.trailers.data.local.converter;

import android.arch.persistence.room.TypeConverter;

import com.an.trailers.data.local.entity.TvEntity;
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.List;

public class TvListTypeConverter {

    @TypeConverter
    public List<TvEntity> fromString(String value) {
        Type listType = new TypeToken<List<TvEntity>>() {}.getType();
        List<TvEntity> tvEntities = new Gson().fromJson(value, listType);
        return tvEntities;
    }

    @TypeConverter
    public String fromList(List<TvEntity> movieEntities) {
        return new Gson().toJson(movieEntities);
    }
}

File: app/src/main/java/com/an/trailers/data/local/converter/TvListTypeConverter.kt
package com.an.trailers.data.local.converter

import android.arch.persistence.room.TypeConverter
import com.an.trailers.data.local.entity.TvEntity
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

class TvListTypeConverter {

    @TypeConverter
    fun fromString(value: String): List<TvEntity>? {
        val listType = object : TypeToken<List<TvEntity>>() {}.type
        return Gson().fromJson<List<TvEntity>>(value, listType)
    }

    @TypeConverter
    fun fromList(list: List<TvEntity>?): String {
        val gson = Gson()
        return gson.toJson(list)
    }
}
